### PR TITLE
Send deep scan discovered pages for scanning 

### PR DESCRIPTION
**Pull request by lamaks**
#### Details

Send deep scan discovered pages for scanning by creating scan batch request DB document.
Process scan batch request document in Cosmos feed trigger.
Expand website scan result provider to accept batch data to update.
Cleanup WebsiteScanResultUpdater from unrelated unused code.

##### Motivation

Feature 1793076: Extend service pipeline to run crawler engine

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] Validated in an Azure resource group

### CODE REVIEW

File: packages/service-library/src/data-providers/website-scan-result-provider.ts [hunk 1]
```diff
@@ -54,16 +74,52 @@ export class WebsiteScanResultProvider {
...
+    }
+
+    private mergeBatch(sourceDocuments: WebsiteScanResult[]): WebsiteScanResult {
+        if (sourceDocuments.length === 1) {
+            return sourceDocuments[0];
+        }
+
+        let targetDocument: WebsiteScanResult = sourceDocuments[0];
+        sourceDocuments.map((sourceDocument) => {
+            targetDocument = this.merge(sourceDocument, targetDocument);
```
Comment (pownkel) [inline packages/service-library/src/data-providers/website-scan-result-provider.ts:105]: Will we need to worry about the possibility of merging several documents with different etags?

Comment (lamaks) [reply packages/service-library/src/data-providers/website-scan-result-provider.ts:105]: No, `mergeOrCreate()` function will handle etag logic of the final document before writing into storage

File: packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                reportGroups: [
+                    {
+                        consolidatedId: websiteScanResult.scanGroupId,
+                    },
+                ],
+            };
+        });
+    }
+
+    private getUrlsToScan(websiteScanResult: WebsiteScanResult): string[] {
```
Comment (pownkel) [inline packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts:96]: This logic is duplicated in discovered-url-processor.ts (called in DeepScanner after crawling)

File: packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                discoveredScanId: scanId,
+                discoveredUrl: url,
+            });
+
+            return {
+                scanId,
+                url,
+                priority: _.isNil(pageScanResult.priority) ? 0 : pageScanResult.priority,
+                deepScan: true,
+                ...(_.isEmpty(pageScanResult.scanNotifyUrl) ? {} : { scanNotifyUrl: pageScanResult.scanNotifyUrl }),
```
Comment (pownkel) [inline packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts:83]: nit: This line is a bit confusing to look at, could it be extracted into a variable?

File: packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts (hunk 2 above)
Comment (lamaks) [reply packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts:96]: Logic is different here. The `DiscoveredUrlProcessor` dedup and limit links from crawl result by comparing with `knownPages`. Here we select links that already dedup in `knownPages` list by comparing with `pageScans` to select links that do not have scan request created yet.

File: packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts (hunk 3 above)
Comment (lamaks) [reply packages/web-api-scan-runner/src/crawl-runner/scan-feed-generator.ts]: updated

**Review by pownkel — APPROVED** [review]
LGTM

**Pull request merged**